wgsl/index.bs

```diff
@@ -7344,6 +7344,9 @@ Note: The [=attribute/const=] attribute cannot be applied to user-declared funct
 * A [=formal parameter|function parameter=] [=shader-creation error|must=] one the following types:
     * a constructible type
     * a pointer type
+        * It [=shader-creation error|must=] be valid to [=variable declaration|declare a variable=]
```

Comment:
Nit: alternative suggested wording, I have a small issue with saying corresponding reference type.

>The [=pointee type=] [shader-creation error|must=] be a valid [=store type=] for a [=variable declaration=] in the same address space.

Reply:
I want to capture the constraints on the access mode too.  I'll edit to simplify.